code/modules/vueui/ui.dm

```diff
@@ -317,17 +324,28 @@ main ui datum.
   * @return 1 if push should happen, 0 if shouldn't happen.
   */
 /datum/vueui/proc/set_status(var/nstatus, var/autopush = TRUE, var/checkforchange = FALSE)
+	. = 0
 	if (nstatus != status) // Only update if it is different
 		status = nstatus
 		if(nstatus > STATUS_DISABLED)
-			return check_for_change(TRUE, !autopush) // Gather data and update it
+			. = check_for_change(TRUE, !autopush) // Gather data and update it
+			if(. == 2)
+				return 1
+			else
+				return 0
 		else if (nstatus == STATUS_DISABLED && autopush)
-			if(autopush) push_change(null) // Only update ui data
-			return 1
+			if(autopush) 
+				push_change(null) // Only update ui data
+			else
+				return 1
 		else
 			close()
 	else if (status > STATUS_DISABLED && checkforchange)
-		return check_for_change(TRUE, !autopush)
+		. = check_for_change(TRUE, !autopush)
+		if(. == 2)
```

Comment:
You can just put the check into the if condition as you overwrite the return value in both cases